### PR TITLE
Add section on cancellation to Go

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Updated synchronous section with more details.

### CODE REVIEW

File: docs/golang/design.md [hunk 1]
```diff
@@ -80,6 +80,8 @@ _Service methods_ are the methods on the client that invoke operations on the se
 
 The Go idiom is to expose only synchronous methods.  This allows callers to implement asynchronous calls as appropriate for their use-case.
 
+{% include requirement/MUSTNOT id="golang-api-sync-async" %} create any goroutines inside an API call or return any channels which implies concurrent behavior.
```
Comment (JeffreyRichter) [inline docs/golang/design.md:83]: Storage does have some helpers that uploads a blob via parallel blocks and this does create goroutines internally to get parallelism. I think this is OK - the goroutines are destroyed once the method returns.

File: docs/golang/design.md [hunk 2]
```diff
@@ -157,7 +159,37 @@ Model structures are types that consumers use to provide required information in
 
 ##### Cancellation
 
-TODO
+Cancellation is handled via the `context.Context` paramater, which is _always_ the first method parameter.  All APIs that perform I/O of any kind, sleep, or perform a significant amount of CPU-bound work _must_ take a `context.Context` as its first parameter.  The following are some examples of how to wait on a context.
+
```
Comment (JeffreyRichter) [inline docs/golang/design.md:163]: You can keep these examples or put a link to context examples from Go. I usually dont' think that we need to repeat whatever is naturally part of the language already.

**Review by JeffreyRichter — APPROVED** [review]

File: docs/golang/design.md (hunk 1 above)
Comment (jhendrixMSFT) [reply docs/golang/design.md:83]: Good point.  I can clarify this to indicate that while the "base" APIs must be synchronous, authors can write higher-level functionality that might not be.

Comment (jhendrixMSFT) [reply docs/golang/design.md]: Updated

File: docs/golang/design.md (hunk 2 above)
Comment (jhendrixMSFT) [reply docs/golang/design.md]: Linked to documentation and removed examples.

**Pull request merged**